src/background/background.ts

```diff
@@ -1,4 +1,151 @@
+March 2017 - January 2023
+As the ex-owner of a top-notch sound studio in Israel, based in Tel Aviv, I led the way on audio design projects for big video production companies. Working with huge names like McDonald's, AMPM, Samsung, Strauss, Tnuva, and providing sound design solutions for big tech companies such as Wix, SAP, Fiverr, Tabula, Simply, Viber, and others. I consistently delivered top-quality work.
+Successfully sold the business for a profit.
+
+
+SKILLS
+Deep understanding of Vanilla Javascript
+fluency with relational and NoSQL databases(MySQL, MongoDB)
+Proficient in implementing best practices for designing and maintaining APIs.
+Experienced in applying responsive design principles for optimal user experience.
+Experience with Git and GitHub
+Experienced in Chrome extension development
+Experienced in Vs Code extension development
+PROGRAMING LANGUAGES
+Javascript, Typescript, C++
+FRAMEWORKS
+Express, ReactJS
+`;
+// Global states - Chrome Storage ---------------------
+
+type GlobalStateKey = 'query'; // we might need:  'jobDescription' | 'currentUrl'  | 'jobPostId';
+
+function setState(key: GlobalStateKey, value: any) {
+  chrome.storage.sync.set({ [key]: value }, () => {
+    console.log(`Data saved - key: ${key} and value: ${value}`);
+  });
+}
+
+function getState(key: GlobalStateKey) {
+  return new Promise((resolve, reject) => {
+    chrome.storage.sync.get(key, (result) => {
+      if (chrome.runtime.lastError) {
+        reject(chrome.runtime.lastError);
+      } else {
+        resolve(result[key]);
+      }
+    });
+  });
+}
+
+function sendToContentScript(message: Message) {
+  chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
+    // Send a message to the content script of the active tab
+    chrome.tabs.sendMessage(
+      tabs[0].id,
+      { message },
+      async (response: Message) => {
+        console.log('Response from content script:', response);
+        if (response.type === 'catch-job-description') {
+          const query = getQuery(response.message);
+          setState('query', query);
+        }
+      }
+    );
+  });
+}
+
+function getQuery(jobDescription: string) {
+  return `You are a smart assistant to career advisors at the Harvard Extension School. Your take is to rewrite
+  resumes to fit a particular job description.
+  
+  Consider the following job description:
+${jobDescription}
+
+and here is the CV:
+${cvText}
+
+Your task is to rewrite the given CV. Follow these guidelines:
+- Be truthful and objective to the experience listed in the CV
+- Be specific rather than general
+- Rewrite job highlight items using STAR methodology (but do not mention STAR explicitly)
+- Fix spelling and grammar errors
+- Write to express not impress
+- Articulate and don't be flowery
+- Prefer active voice over passive voice
+- Do not include a summary about the candidate
+
+Improved CV:
+`;
+}
+
+// types
+type MessageType = 'catch-job-description';
+export interface Message {
```

Comment:
since this is also used by the content script, better place the `shared types` in their own separate file